src/compiler/commandLineParser.ts

```diff
@@ -405,13 +405,107 @@ namespace ts {
       */
     export function parseConfigFileTextToJson(fileName: string, jsonText: string): { config?: any; error?: Diagnostic } {
         try {
-            return { config: /\S/.test(jsonText) ? JSON.parse(jsonText) : {} };
+            let jsonTextWithoutComments = removeComments(jsonText);
+            return { config: /\S/.test(jsonTextWithoutComments) ? JSON.parse(jsonTextWithoutComments) : {} };
         }
         catch (e) {
             return { error: createCompilerDiagnostic(Diagnostics.Failed_to_parse_file_0_Colon_1, fileName, e.message) };
         }
     }
 
+
+    /**
+     * Remove the comments from a json like text.
+     * Comments can be single line comments (starting with # or //) or multiline comments using / * * /
+     *
+     * This method replace comment content by whitespace rather than completely remove them to keep positions in json parsing error reporting accurate.
+     */
+    function removeComments(jsonText: string): string {
+        let result = "";
+        let processingString = false;
+        let processingSingleLineComment = false;
+        let processingMultiLineComment = false;

```

Comment:
I would rather see these booleans as a single enum (along with the base parser state), since only one of them can be selected at a time. It would also be nice if the parsing actions associated with each state were isolated into functions, rather than left as conditional branches.

So:
```ts
let i = 0;
for (; i < jsonText.length; i++) {
  let currentCharCode = jsonText.charCodeAt(i);
  let currentChar = jsonText.charAt(i);
  let nextCharCode = (i + 1 < jsonText.length) ? jsonText.charCodeAt(i + 1) : undefined;
  switch (parserState) {
    case JSONCommentParserState.String:
      parserState = parseString(currentCharCode, currentChar, nextCharCode);
      break;
    case JSONCommentParserState.SingleLineComment:
      parserState = parseSingleLineComment(currentCharCode, currentChar, nextCharCode);
      break;
    // ... and so on
  }
}

function parseString(currentCharCode: number, currentChar: string, nextCharCode: number) {
  // ... contents of `if (processingString) {` branch
}
```